src/utils/deserializeSearchResponse.ts

```diff
@@ -9,12 +9,12 @@ const convertMetadataToObject = (metadata: DocMetadata[]) => {
   return obj;
 };
 
-const parseMetadata = (rawMetadata: DocMetadata[]) => {
+const parseMetadata = (rawMetadata: DocMetadata[], matching_text: string) => {
```

Comment:
For maintainability I suggest keeping casing consistent, e.g. changing `matching_text` to `matchingText`.